win32ss/user/ntuser/winpos.c

```diff
@@ -3709,35 +3709,51 @@ NtUserSetWindowPlacement(HWND hWnd,
    TRACE("Enter NtUserSetWindowPlacement\n");
    UserEnterExclusive();
 
-   if (!(Wnd = UserGetWindowObject(hWnd)) ||
-        UserIsDesktopWindow(Wnd) || UserIsMessageWindow(Wnd))
-   {
-      goto Exit; // Return FALSE
-   }
+    Wnd = UserGetWindowObject(hWnd);
+    if (!Wnd)
+        goto Exit; // Return FALSE
+
+    UserRefObjectCo(Wnd, &Ref);
+    if (UserIsDesktopWindow(Wnd) || UserIsMessageWindow(Wnd))
+    {
+        UserDerefObjectCo(Wnd);
+        goto Exit; // Return FALSE
+    }
 
    _SEH2_TRY
    {
       ProbeForRead(lpwndpl, sizeof(WINDOWPLACEMENT), 1);
-      RtlCopyMemory(&Safepl, lpwndpl, sizeof(WINDOWPLACEMENT));
+      Safepl = *lpwndpl;
    }
    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
    {
+      UserDerefObjectCo(Wnd);
       SetLastNtError(_SEH2_GetExceptionCode());
       _SEH2_YIELD(goto Exit); // Return FALSE
    }
    _SEH2_END
 
-   if(Safepl.length != sizeof(WINDOWPLACEMENT))
-   {
-      goto Exit; // Return FALSE
-   }
+    /* Backwards-compatibility: NT3.x doesn't check the length */
+    if (Safepl.length == sizeof(WINDOWPLACEMENT) ||
+#if 1 // FIXME: CORE-19675: We have to populate dwExpWinVer
```

Comment:
do you plan to fix that one first before merging this PR?

Reply:
#7095